.github/workflows/iroha2-dev.yml

```diff
@@ -16,10 +15,10 @@ env:
 # no point in burning our AWS self-hosted runners' time. Hence
 # `ubuntu-latest` and not `[self-hosted, Linux]`.
 jobs:
-  dockerhub:
+  registry:
     runs-on: [self-hosted, Linux, iroha2-dev-push]
     container:
-      image: 7272721/i2-ci:nightly
+      image: hyperledger/iroha2-ci:nightly-2022-12-22
```

Comment:
Just FYI for @mversic, @Arjentix, @Erigara, @ilchu and @pesterev. The decision to hard-code these numbers is to ensure three things: 
- No extra information is needed in `iroha2-stable` or `iroha2-lts` in order to open PRs into those branches. Not really relevant given the backport deprecation, but it will be useful after `2.0` release. 
- The image used in CI can be paired to the correct toolchain version, so a simple `grep` can answer the same question that used to necessitate the `rust-toolchain.toml` file, which caused *many* issues with both CI and build. 
- Replacing the toolchain can be done with a verbatim find and replace of `nightly-202X-XX-XX` with `nightly-202Y-YY-YY`.